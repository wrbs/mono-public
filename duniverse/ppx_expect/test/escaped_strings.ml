let%expect_test "escaped carriage return" =
  print_string "a\r\nb";
  [%expect " \n a\n b\n "];
  print_string "a\r\nb";
  [%expect_exact "a\r\nb"]
;;

let%expect_test "escaped tab" =
  print_string "a\tb";
  [%expect "a\tb"];
  print_string "a\tb";
  [%expect_exact "a\tb"]
;;

let%expect_test "escaped quote" =
  print_string "a\"b";
  [%expect "a\"b"];
  print_string "a\"b";
  [%expect_exact "a\"b"]
;;

let%expect_test "escaped trailing carriage return" =
  print_string "a\r\nb\r\n";
  [%expect " \n a\n b\n "];
  print_string "a\r\nb\r\n";
  [%expect_exact "a\r\nb\r\n"]
;;
