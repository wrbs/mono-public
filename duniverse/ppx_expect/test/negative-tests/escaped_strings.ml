let%expect_test "escaped carriage return" =
  print_string "a\rb";
  [%expect ""];
  print_string "a\rb";
  [%expect_exact ""];
  print_string "a\r\nb";
  [%expect ""];
  print_string "a\r\nb";
  [%expect_exact ""];
  print_string "a\n\rb";
  [%expect ""];
  print_string "a\n\rb";
  [%expect_exact ""]
;;

let%expect_test "escaped tab" =
  print_string "a\tb";
  [%expect ""];
  print_string "a\tb";
  [%expect_exact ""];
  print_string "a\t\nb";
  [%expect ""];
  print_string "a\t\nb";
  [%expect_exact ""];
  print_string "a\n\tb";
  [%expect ""];
  print_string "a\n\tb";
  [%expect_exact ""]
;;

let%expect_test "escaped quote" =
  print_string "a\"b";
  [%expect ""];
  print_string "a\"b";
  [%expect_exact ""]
;;

let%expect_test "escaped trailing carriage return" =
  print_string "a\r\nb\r\n";
  [%expect ""];
  print_string "a\r\nb\r\n";
  [%expect_exact ""];
  print_string "a\r\nb\r\n";
  [%expect];
  print_string "a\r\nb\r\n";
  [%expect_exact]
;;

let%expect_test "unescaped carriage return --- empty expect" =
  print_string "a\r\nb";
  [%expect];
  print_string "a\r\nb";
  [%expect_exact]
;;

let%expect_test "unescaped carriage return --- populated expect" =
  print_string "a\r\nb";
  [%expect {|
    a
    b |}];
  print_string "a\r\nb";
  [%expect_exact {|a
b|}]
;;
