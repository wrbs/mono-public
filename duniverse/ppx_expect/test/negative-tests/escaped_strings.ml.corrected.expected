let%expect_test "escaped carriage return" =
  print_string "a\rb";
  [%expect "a\rb"];
  print_string "a\rb";
  [%expect_exact "a\rb"];
  print_string "a\r\nb";
  [%expect " \n a\n b\n "];
  print_string "a\r\nb";
  [%expect_exact "a\r\nb"];
  print_string "a\n\rb";
  [%expect " \n a\n b\n "];
  print_string "a\n\rb";
  [%expect_exact "a\n\rb"]
;;

let%expect_test "escaped tab" =
  print_string "a\tb";
  [%expect "a\tb"];
  print_string "a\tb";
  [%expect_exact "a\tb"];
  print_string "a\t\nb";
  [%expect " \n a\n b\n "];
  print_string "a\t\nb";
  [%expect_exact "a\t\nb"];
  print_string "a\n\tb";
  [%expect " \n a\n b\n "];
  print_string "a\n\tb";
  [%expect_exact "a\n\tb"]
;;

let%expect_test "escaped quote" =
  print_string "a\"b";
  [%expect "a\"b"];
  print_string "a\"b";
  [%expect_exact "a\"b"]
;;

let%expect_test "escaped trailing carriage return" =
  print_string "a\r\nb\r\n";
  [%expect " \n a\n b\n "];
  print_string "a\r\nb\r\n";
  [%expect_exact "a\r\nb\r\n"];
  print_string "a\r\nb\r\n";
  [%expect
    {|
    a
    b
    |}];
  print_string "a\r\nb\r\n";
  [%expect_exact
    {|a
b
|}]
;;

let%expect_test "unescaped carriage return --- empty expect" =
  print_string "a\r\nb";
  [%expect
    {|
    a
    b
    |}];
  print_string "a\r\nb";
  [%expect_exact
    {|a
b|}]
;;

let%expect_test "unescaped carriage return --- populated expect" =
  print_string "a\r\nb";
  [%expect
    {|
    a
    b
    |}];
  print_string "a\r\nb";
  [%expect_exact
    {|a
b|}]
;;
